android-core/plugins/org.eclipse.andmore/src/org/eclipse/andmore/internal/build/builders/BaseBuilder.java

```diff
@@ -320,10 +320,16 @@ protected void abortOnBadSetup(@NonNull IJavaProject javaProject,
         if (projectState == null) {
             projectState = Sdk.getProjectState(javaProject.getProject());
         }
-
+        
+        // the project state can still be null, 
+        // e.g. if there is no project.properties file
+        if (projectState == null) {
```

Comment:
@locosmac Appreciate the time you have taken to address these items.  I'll see about getting some of these merged tonight.  Can you please open a separate bug for the issue you are experiencing with the integration test plugins compiling.  You shouldn't have to spend time getting those fixed.